src/compiler/msvc.rs

```diff
@@ -503,18 +503,94 @@ pub fn parse_arguments(
     let mut pdb = None;
     let mut depfile = None;
     let mut show_includes = false;
+    let mut has_pending_clang_arguments = false;
     let mut xclangs: Vec<OsString> = vec![];
     let mut clangs: Vec<OsString> = vec![];
     let mut profile_generate = false;
     let mut multiple_input = false;
     let mut multiple_input_files = Vec::new();
 
+    macro_rules! parse_clang_arguments {
```

Comment:
why is it a macro and not a function? thanks

Reply:
Because this code uses the `try_or_cannot_cache!` and `cannot_cache!` macros that are intended to make `parse_arguments` return early, it cannot easily be refactored into a function.